app/components/signature-pill.js

```diff
@@ -11,84 +12,53 @@ export default class SignaturePillComponent extends Component {
-    }
-  }
-
-  get title() {
-    if (this.isCancelled) {
-      return this.intl.t('cancelled');
-    } else if (this.isRefused) {
-      return this.intl.t('refused');
-    } else if (this.isSigned) {
-      return this.intl.t('signed');
-    } else if (this.hasToBeSigned) {
-      return this.intl.t('to-sign');
-    } else if (this.hasToBeApproved) {
-      return this.intl.t('to-approve');
-    } else if (this.isPrepared) {
-      return this.intl.t('sent');
-    } else if (this.isMarked) {
-      return this.intl.t('to-sign');
-    }
-    return "";
-  }
-
   loadData = task(async () => {
+    const { SIGNED, REFUSED } = constants.SIGNFLOW_STATUSES;
     const signMarkingActivity = await this.args.signMarkingActivity;
-    if (signMarkingActivity) {
-      const signSubcase = await signMarkingActivity.signSubcase;
-      const signPreparationActivity = await signSubcase.signPreparationActivity;
-      const signApprovalActivities = await signSubcase.signApprovalActivities;
-      const signSigningActivities = await signSubcase.signSigningActivities;
-      const signCompletionActivity = await signSubcase.signCompletionActivity;
-      const signRefusalActivities = await signSubcase.signRefusalActivities;
-      const signCancellationActivity = await signSubcase.signCancellationActivity;
-
-      this.isMarked = !!signMarkingActivity;
-      this.isPrepared = !!signPreparationActivity;
-      this.hasToBeApproved = signApprovalActivities?.some((activity) => activity.startDate && !activity.endDate);
-      this.hasToBeSigned = signSigningActivities?.some((activity) => activity.startDate && !activity.endDate);
-      this.isSigned = !!signCompletionActivity;
-      this.isRefused = signRefusalActivities?.length;
-      this.isCancelled = !!signCancellationActivity;
+    if (!signMarkingActivity) return;
+    const signSubcase = await signMarkingActivity.signSubcase;
+    const signFlow = await signSubcase.signFlow;
+    const status = await signFlow.status;
+    let signingHubUrl = null;
 
-      if (!this.isRefused) {
-        const piece = await this.args.piece;
-        const signFlow = await signSubcase.signFlow;
-        const signFlowCreator = await signFlow.creator;
-        const currentUser = this.currentSession.user;
-        if (piece && (signFlowCreator.id === currentUser.id) && !this.isSigned) {
-          const response = await fetch(
-            `/signing-flows/${signFlow.id}/pieces/${piece.id}/signinghub-url?collapse_panels=false`
-          );
-          if (response.ok) {
-            const result = await response.json();
-            this.signingHubUrl = result.url;
-          } 
+    if (status.uri !== REFUSED) {
+      const piece = await this.args.piece;
+      const signFlow = await signSubcase.signFlow;
+      const signFlowCreator = await signFlow.creator;
+      const currentUser = this.currentSession.user;
+      if (
+        piece &&
+        signFlowCreator.id === currentUser.id &&
+        status.uri !== SIGNED
```

Comment:
Found while optimizing another PR. It would be nice if we check for `MARKED` status here as well. Signflows with that status will never have a SigningHub URL, so the call will always fail here.